modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx

```diff
@@ -0,0 +1,104 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import { useState } from "react";
+
+import { Typography, withStyles } from "@material-ui/core";
+
+import PropTypes from "prop-types";
+
+import MultipleChoice from "./MultipleChoice";
+import Question from "./Question";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+function NumberQuestion(props) {
+  let {defaults, max, min, name, userInput, minValue, maxValue, type, errorText, ...rest} = props;
+  const [error, setError] = useState(false);
+
+  // Callback function for our min/max
+  let checkNumber = (text) => {
+    let value = 0;
+    if (type === "integer") {
+      // Test that it is an integer
+      if (!/^[-+]?\d*/.test(text)) {
+        setError(true);
+        return;
+      }
+
+      value = parseInt(text);
+    } else if (type === "float") {
+      // Test that it can be parsed as a float
+      if (isNaN(text) || !/^[-+]?\d*([,.]\d+)?/.test(text)) {
+        setError(true);
+        return;
+      }
+
+      value = parseFloat(text);
```

Comment:
Also missing a `$` at the end, without it it will happily accept any text.

Things I don't like:

- `-` and `+` are accepted
- empty string is accepted
- `2,2` is accepted here, but will later turn into `2` because `parseFloat` stops at the comma

Luckily, almost all shortcomings of the regexp are nullified by the `isNaN` check, which rejects all except the empty string.

So, how about this shorter check?

```suggestion
      value = Number(text);
      // Reject whitespace and non-numbers
      if (/^\s*$/.test(text) || isNaN(value)) {
        setError(true);
        return;
      }
```

Reply:
The intention of allowing commas as in `2,2` was to allow commas used as decimal points, as may be the case in other countries, but it seems like `parseFloat` assumes decimal points anyway.